music21/opensheetmusicdisplay/osmd.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#-------------------------------------------------------------------------------
+# Name:         opensheetmusicdisplay/osmd.py
+# Purpose:      music21 classes for displaying music21 scores in IPython notebooks
+#
+# Authors:      Sven Hollowell
+#
+# Copyright:    Copyright © 2012-14 Michael Scott Cuthbert and the music21 Project
+# License:      LGPL or BSD, see license.txt
+#-------------------------------------------------------------------------------
+'''
+Registers the .show('osmd') converter for use in IPython browser
+See OSMD project page here: https://github.com/opensheetmusicdisplay/opensheetmusicdisplay
+'''
+import unittest
+from music21 import exceptions21
+
+
+class OpenSheetMusicDisplayException(exceptions21.Music21Exception):
+    pass
+
+hasInstalledIPython = False
+try:
+    # raise ImportError('for testing')
+    from IPython.core.display import display, HTML, Javascript
+    hasInstalledIPython = True
+except ImportError as e:
+    # raise error only if methods are called by a .show()
+    def display(*args, **kwargs):
+        raise OpenSheetMusicDisplayException('OpenSheetMusicDisplay requires IPython to be installed')
+    HTML = Javascript = display
+
+
+from music21.converter.subConverters import SubConverter
+from music21.instrument import Piano
+from urllib.request import pathname2url
+import time, random, json, os
+
+
+class ConverterOpenSheetMusicDisplay(SubConverter):
+    '''
+    Converter for displaying sheet music inline
+    in an IPython notebook.
+    Uses: https://opensheetmusicdisplay.org/ for rendering
+    '''
+    registerFormats = ('osmd',)
+    registerShowFormats = ('osmd',) 
+
+    def show(self, obj, fmt, 
+            fixPartName=True, offline=True, div_id=None,
```

Comment:
Feel free to add "*' before fixPartName -- the rest should be keyword only.

Reply:
`div_id` should be `divId` -- sorry that music21 was originally a Perl package and we've stuck with CamelCase variables throughout.  It's okay to use PEP8 underscore spacing in internal variables, but since it's exposed in the arguments, it should conform to the music21 convention (which has worked out well for music21j, since Javascript uses camelcase conventions)